src/org/pentaho/reporting/platform/plugin/ReportContentUtil.java

```diff
@@ -106,13 +106,27 @@ public static ValidationResult applyInputsToReportParameters( final MasterReport
 
   public static Object computeParameterValue( final ParameterContext parameterContext,
       final ParameterDefinitionEntry parameterDefinition, final Object value ) throws ReportProcessingException {
-    if ( value == null ) {
+      Object def = null;
+      if ( value == null ) {
       // there are still buggy report definitions out there ...
-      return null;
+      if ( parameterDefinition.getDefaultValue( parameterContext ) != null ) {
+        def = parameterDefinition.getDefaultValue( parameterContext );
+      } else {
+        return null;
+      }
     }
 
     final Class valueType = parameterDefinition.getValueType();
     final boolean allowMultiSelect = isAllowMultiSelect( parameterDefinition );
+
```

Comment:
Isn't this too strong ? Shouldn't this be done only when we know the parameter is of type List ?
Otherwise, we're converting every string with a comma to an array list value which could be wrong.

I could be wrong but that's what it seems like it's being done here.